ivy_tests/test_ivy/test_frontends/test_numpy/test_utility_routines/test_matrix_any.py

```diff
@@ -0,0 +1,124 @@
+# global
+import numpy as np
+from hypothesis import given, strategies as st
+
+# local
+import ivy
+import ivy_tests.test_ivy.helpers as helpers
+import ivy_tests.test_ivy.test_frontends.test_numpy.helpers as np_frontend_helpers
+
+
+@st.composite
+def _array_with_dtype_axis_keepdims_and_where(draw):
+    dtypes = draw(
+        helpers.array_dtypes(
+            num_arrays=1
+        )
+    )
+    shape = draw(
+        helpers.get_shape(
+            min_num_dims=1,
+            max_num_dims=5
+        )
+    )
+    axis = draw(
+        helpers.ints(min_value=-1, max_value=len(shape) - 1)
+    )
+    if axis == -1:
+        axis = None
+    x = draw(
+        helpers.array_values(
+            shape=shape,
+            dtype=dtypes[0]
+        )
+    )
+    where_shape_length = draw(helpers.ints(min_value=0, max_value=len(shape)))
+    if where_shape_length != 0:
+        where_nb_dims_to_change = draw(
+            helpers.ints(
+                min_value=0,
+                max_value=where_shape_length - 1
+            )
+        )
+        where_dims_to_change = [draw(
+            helpers.ints(
+                min_value=0,
+                max_value=where_shape_length - 1
+            )) for i in range(where_nb_dims_to_change)
+        ]
+        where_dims_list = [1] * where_shape_length
+        for dim in where_dims_to_change:
+            where_dims_list[dim] = shape[::-1][dim]
+        where_dims_list = where_dims_list[::-1]
+        where = draw(
+            helpers.array_values(
+                shape=where_dims_list,
+                dtype="bool"
+            )
+        )
+    else:
+        where = draw(
+            st.booleans()
+        ) 
+    keepdims = draw(
+        st.booleans()
+    )
+    return x, dtypes[0], axis, keepdims, where
+
+
+@given(
+    x_dtype_axis_keepdims_where=_array_with_dtype_axis_keepdims_and_where(),
```

Comment:
The issue seems to be originating from this line. Checkout the stacktrace from here: https://github.com/unifyai/ivy/runs/8255557671?check_suite_focus=true#step:6:13256